tensorpipe/test/channel/channel_test.h

```diff
@@ -19,6 +19,40 @@
 #include <tensorpipe/test/peer_group.h>
 #include <tensorpipe/transport/uv/context.h>
 
+class DataWrapper {
+ public:
+  virtual ~DataWrapper() = default;
+  virtual void* data() = 0;
+  virtual size_t size() = 0;
+  virtual void wrap(const void* ptr) = 0;
+  virtual void unwrap(void* ptr) = 0;
+};
+
+class IdWrapper : public DataWrapper {
+ public:
+  IdWrapper(size_t len) : size_(len) {}
+
+  void* data() override {
+    return data_;
+  }
+
+  size_t size() override {
+    return size_;
+  }
+
+  void wrap(const void* ptr) override {
+    data_ = const_cast<void*>(ptr);
+  }
+
+  void unwrap(void* ptr) override {
+    ASSERT_EQ(data_, ptr);
+  }
```

Comment:
It seems currently we only ever wrap immediately after the constructor and we unwrap just before the destructor: would it make sense to avoid having explicit methods to do the (un)wrapping and do it in the (de/con)structor? Also, will it ever happen that unwrap is called with a pointer different than the one in wrap? If not, maybe we can just remove the argument from unwrap?

Reply:
In the case of CUDA buffers, unwrapping implies a `cudaMemcpy`, and relying on it being performed implicitly as part of destroying the wrapper didn't seem right.

Reply:
Ok, must have missed that, it's fine then